core/src/main/scala/akka/persistence/cassandra/query/EventsByTagPublisher.scala

```diff
@@ -280,6 +282,9 @@ private[query] class EventsByTagPublisher(
       } else {
         if (backtrackingMode != NoBacktracking)
           currTimeBucket = TimeBucket(currOffset)
+        // make sure LookingForDelayed is not triggered immediately again in case of slow queries
+        if (backtrackingMode == LookingForDelayed)
+          lookForDelayedDeadline = nextLookForDelayedDeadline()
```

Comment:
I wish we could refactor this actor to be less `if a ..., if b ..., if else if if`.  Maybe outside the scope of this PR but I find it makes me need to keep more things in my head than I want to when reading these sources.